<launch>
  <node pkg="rs4se"
      type="intel_d435i"
      name="rs"
      required="true"
      output="screen">
    <!-- ROS TOPICS -->
    <param name="rgb0_topic" type="str" value="/rgb0/image" />
    <param name="ir0_topic" type="str" value="/ir0/image" />
    <param name="ir1_topic" type="str" value="/ir1/image" />
    <param name="depth0_topic" type="str" value="/depth0/image" />
    <param name="gyro0_topic" type="str" value="/gyro0/data" />
    <param name="accel0_topic" type="str" value="/accel0/data" />
    <param name="imu0_topic" type="str" value="/imu0/data" />

    <!-- INTEL REALSENSE -->
    <param name="global_time" type="bool" value="true" />
    <!-- RGBD -->
    <param name="enable_rgb" type="bool" value="true" />
    <param name="enable_ir" type="bool" value="true" />
    <param name="enable_depth" type="bool" value="true" />
    <param name="enable_emitter" type="bool" value="true" />
    <param name="rgb_width" value="640"/>
    <param name="rgb_height" value="480"/>
    <param name="rgb_frame_rate" value="30"/>
    <param name="rgb_format" type="str" value="RGB8" />
    <param name="rgb_exposure" type="double" value="100" />
    <param name="ir_width" type="int" value="640" />
    <param name="ir_height" type="int" value="480" />
    <param name="ir_format" type="str" value="Y8" />
    <param name="ir_frame_rate" type="int" value="30" />
    <param name="ir_exposure" type="double" value="10000" />
    <param name="depth_width" type="int" value="640" />
    <param name="depth_height" type="int" value="480" />
    <param name="depth_format" type="str" value="Z16" />
    <param name="depth_frame_rate" type="int" value="30" />
    <!-- MOTION -->
    <param name="enable_motion" type="bool" value="true" />
    <param name="gyro_hz" type="double" value="400" />
    <param name="accel_hz" type="double" value="250" />
  </node>

  <!-- Disable image transport plugins -->
  <group ns="rs/rgb0/image" >
    <rosparam param="disable_pub_plugins">
      - "image_transport/compressed"
      - "image_transport/compressedDepth"
      - "image_transport/theora"
    </rosparam>
  </group>

  <group ns="rs/ir0/image" >
    <rosparam param="disable_pub_plugins">
      - "image_transport/compressed"
      - "image_transport/compressedDepth"
      - "image_transport/theora"
    </rosparam>
  </group>

  <group ns="rs/ir1/image" >
    <rosparam param="disable_pub_plugins">
      - "image_transport/compressed"
      - "image_transport/compressedDepth"
      - "image_transport/theora"
    </rosparam>
  </group>

  <group ns="rs/depth0/image" >
    <rosparam param="disable_pub_plugins">
      - "image_transport/compressed"
      - "image_transport/compressedDepth"
      - "image_transport/theora"
    </rosparam>
  </group>

</launch>
